Add option to not charge sales tax

becdetat opened the pull request:
Fixes #3456

This adds an option to not charge sales tax to the onboarding tax task (both with Jetpack/WCS connected and without).

### Detailed test instructions:

Without Jetpack/WCS connected:

- Set up a test site with this plugin installed
- Go through the OBW, don't set up WCS or Jetpack
- Enter the "Set up tax" task
- The "My business doesn't charge sales tax" option should be displayed per below:
![image](https://user-images.githubusercontent.com/224531/92544808-1afa9c00-f292-11ea-92dd-a8f3f572abf8.png)
- Select "My business doesn't charge sales tax"
- You should be redirected to the home page and the "Set up tax" task should be marked as complete

With Jetpack/WCS connected:

- Set up a test site with this plugin installed
- Go through the OBW, set up WCS and Jetpack
- Make sure Jetpack is connected
- Enter the "Set up tax" task
- The "My business doesn't charge sales tax" option should be displayed per below:
![image](https://user-images.githubusercontent.com/224531/92545048-aaa04a80-f292-11ea-864c-b940d2985bbc.png)
- Select "My business doesn't charge sales tax"
- You should be redirected to the home page and the "Set up tax" task should be marked as complete

### Changelog Note:

Add: Option to not charge sales tax

becdetat committed:
Add option to not set up sales tax

jeffstieler reviewed (changes requested):
Looks like some linter errors need to get cleaned up to fix CI: https://travis-ci.com/github/woocommerce/woocommerce-admin/jobs/382580088#L862

The code here looks good otherwise and tested well for the scenarios described.

We should add the "I don't charge tax" button to the `plugins` step as well. In my testing I ran into this:
![Screen Shot 2020-09-09 at 10 12 00 AM](https://user-images.githubusercontent.com/63922/92614393-fd293900-f289-11ea-9e80-8628482e02dc.png)

becdetat committed:
Correct JS lint errors

becdetat committed:
Add abort option to plugin step

becdetat committed:
Inline redirect to home page

becdetat commented:
Thanks @jeffstieler 

I've fixed the lint errors, strange that they didn't get picked up on commit, I thought we ran JS lint locally.

> We should add the "I don't charge tax" button to the plugins step as well

Done. I also added a couple of tests to cover the new 'abort' button in `Connect` and `Plugins`.

jeffstieler commented on client/dashboard/components/connect/test/index.js:
```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
```
Hooray tests!

AFAIK we've made the decision to write new tests using `testing-library/react`, not Enzyme. Gutenberg has made this move, as well as other teams at a8c - though I can't find anything "official" on our P2.

This is a rather small test, so I'll consider the rewrite optional for landing this PR. I'll create a follow up issue to migrate our Enzyme tests as well.

timmyc replied:
Thanks for creating an issue to migrate the enzyme tests Jeff - seems like a great cooldown task.

becdetat replied:
I've changed this test to use `testing-library/react` (07a74620) - I like it :-) however I didn't migrate the other test because I don't want to migrate the entire suite in this PR.

jeffstieler commented on client/task-list/tasks/tax.js:
```diff
@@ -167,6 +167,21 @@ class Tax extends Component {
+		} );
+
+		updateOptions( {
+			woocommerce_no_sales_tax: true,
```
Looks like we should also set `woocommerce_calc_taxes` to `no` in this scenario.

becdetat replied:
Nice, done in 266be839

jeffstieler reviewed (changes requested):
This is super close! I think after turning off tax calculation for the store when going through the "abort" flow this will be good to go.

becdetat committed:
Use testing-library/react for the new test

becdetat committed:
Also set woocommerce_calc_taxes to 'no'

jeffstieler reviewed (approved):
🚢  it!
